### ISSUE TITLE
`SelectionArea` cursor should only change when over text

### ISSUE THREAD

**Issue by guidezpl** [flutter/flutter#104594]
Great feature, just found a couple of minor bugs.

## Expected

The cursor should only change when over text.

## Actual
Currently, when the cursor is over a selectable widget such as `Text`, the cursor changes to a text selection cursor over the entire area taken by the widget. 

Take the following for example:
```
import 'package:flutter/material.dart';

class SelectionAreaExpandedRepro extends StatelessWidget {
  const SelectionAreaExpandedRepro({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: SelectionArea(
        child: Scaffold(
          body: Row(
            children: const [
              Expanded(
                child: Text('Hello'),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

https://user-images.githubusercontent.com/6655696/170216883-54cf6d11-a62c-4cca-86bc-89b50782933b.mp4

This also applies vertically.

https://user-images.githubusercontent.com/6655696/170216881-80a459f1-3621-4096-a5bc-3b5c9309ce9c.mp4

**Comment by nilsreichardt** [comment]
Same applies for `ListTile`:

https://user-images.githubusercontent.com/24459435/170222583-1dba2b62-8c85-44b9-814a-830363f35be6.mov

<details>
<summary>My Code</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: SelectionArea(
        child: Scaffold(
          body: Padding(
            padding: EdgeInsets.all(16),
            child: ListTile(
              title: Text('You are cool!'),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

**Comment by HansMuller** [comment]
CC @chunhtai

**Comment by justinmc** [comment]
Seems unrelated but this reminds me of https://github.com/flutter/flutter/issues/131435.

**Comment by gnprice** [comment]
Fixed by #140621.
